File: python programs/chef_and_periodic.py
t = int(input())
for y in range(t):
	n = int(input())
	l1 = list(map(lambda x : int(x), input().split()))
	count = 0
	length = 0
	temp = -1
	for i in range(len(l1)):
		if l1[i] == -1:
			count += 1
			if temp != -1:
				temp += 1
		else:
			if i == 0 or temp == -1:
				temp = l1[i]
			else:
				if l1[i] == (temp + 1):
					temp += 1
				else:
					break
		length += 1

	if count == n:
		print("inf")
	elif length == 1:
		print('impossible')
	else:
		print(length)

File: c++ programs/chef_and_periodic.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	int t;
	cin>>t;
	while(t--){
	long long int i,n,count = 0;
	cin>>n;
	long long int a[n],max, temp;
	for(i=0;i<n;i++){
	    cin>>a[i];
	    if(a[i] == -1){
	        count++;
	        a[i] = a[i-1]+1;
	    }
	}
	if(count == n){
	    cout<<"inf";
	    goto h;
	}
	max = *max_element(a, a+n);
	temp = a[0];
	for(i=1;i<n;i++){
	    if(temp+1 != a[i] && a[i]!=1){
	        cout<<"impossible";
	        goto h;
	    }
	    temp = a[i];
	}
	cout<<max;
	h: cout<<endl;
	}
	return 0;
}